lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -217,6 +198,8 @@ public function getMetadataFor($className)
      */
     public function hasMetadataFor($className)
     {
+        $className = $this->getRealClassName($className);
```

Comment:
Please revert this one: overhead should stay very minimal here, and indeed, duplicates in `$this->loadedMetadata` are expected